app/views/govuk_publishing_components/components/_layout_footer.html.erb

```diff
@@ -26,25 +29,40 @@
               # If the list has multiple columns and there is only one link.
               # This is to prevent a long link wrapping in a column, which
               # leaves an obvious blank space to the right.
-              single_item_list = (( item[:columns] == 2 || item[:columns] == 3 ) && item[:items].length == 1 )
+              single_item_list = (( section[:columns] == 2 || section[:columns] == 3 ) && section[:items].length == 1 )
 
               list_classes = %w[govuk-footer__list]
-              list_classes << "govuk-footer__list--columns-#{item[:columns]}" if item[:columns] unless single_item_list
+              list_classes << "govuk-footer__list--columns-#{section[:columns]}" if section[:columns] unless single_item_list
             %>
             <div class="<%= width_class %> govuk-!-display-none-print">
-              <h2 class="govuk-footer__heading govuk-heading-m"><%= item[:title] %></h2>
+              <h2 class="govuk-footer__heading govuk-heading-m"><%= section[:title] %></h2>
                 <ul class="<%= list_classes.join(' ') %>">
-                  <% item[:items].each do |item| %>
+                  <% section[:items].each_with_index do |item, index| %>
                     <% if item[:href] && item[:text] %>
                       <li class="govuk-footer__list-item">
                         <%
                           attributes = {
                             class: "govuk-footer__link",
                             data: {
-                              "track-category": "footerClicked"
+                              "track-category": "footerClicked",
                             }
-                          }.merge(item.fetch(:attributes, {}))
+                          }
+                          attributes = attributes.merge(item.fetch(:attributes, {}))
                           attributes[:rel] = "noopener" if attributes[:target] == "_blank" && !attributes[:rel]
+
+                          unless attributes[:data][:ga4_link]
```

Comment:
This section seems to be nearly repeated within the template, could it be moved into the helper? You could write a method to call it with the varying parameters e.g. `footer_helper.get_ga4_attributes((index + 1).to_s, (ga4_section_index + 1).to_s)`.

Reply:
Thanks, moved to a helper method.